newrelic-agent/src/main/java/com/newrelic/agent/InitProblemClasses.java

```diff
@@ -73,6 +73,15 @@ public static final void loadInitialClasses() {
             } catch (Throwable e) {
                 // This might happen on non-sun JVMs
             }
+
+            // This works around a ClassCircularityError with the doubly nested inner of ReentrantReadWriteLock.
+            try {
+                String holdCounterClassName = "java.util.concurrent.locks.ReentrantReadWriteLock$Sync$HoldCounter";
+                Class.forName(holdCounterClassName);
+                Agent.LOG.log(Level.FINE, "Worked around loading class " + holdCounterClassName);
```

Comment:
We probably don't need to log here.

Reply:
Agreed, but I kept it at `FINE` to not cloud the logs.  Since it only happens once at startup, probably ok right?  I can remove it if you think it's unhelpful.

Reply:
I like the log